### PR TITLE
add speed name arg

### PR DESCRIPTION
**Pull request by raybellwaves**
Closes https://github.com/anitagraser/movingpandas/issues/134

### CODE REVIEW

File: movingpandas/trajectory.py [hunk 1]
```diff
@@ -247,7 +246,7 @@ def get_speed_column_name(self):
         -------
         string
         """
-        return SPEED_COL_NAME
+        return self.speed_col_name
```
Comment (raybellwaves) [inline movingpandas/trajectory.py:249]: This is tricky as it only exists if you run `.add_speed()`

**Comment by raybellwaves** [comment]
Still a few test failures

test_trajectory_collection.py .............F.FF.......                                                                                            [ 81%]
test_trajectory_plotter.py FF                                                                                                                     [ 85%]
test_trajectory_splitter.py ........FF.....

**Comment by raybellwaves** [comment]
Tests now pass locally

test_geometry_utils.py ....................                                                                                                       [ 11%]
test_overlay.py ...............                                                                                                                   [ 20%]
test_point_clusterer.py ....                                                                                                                      [ 22%]
test_trajectory.py .................................................................                                                              [ 61%]
test_trajectory_aggregator.py ..........                                                                                                          [ 67%]
test_trajectory_collection.py ........................                                                                                            [ 81%]
test_trajectory_generalizer.py .....                                                                                                              [ 84%]
test_trajectory_plotter.py ..                                                                                                                     [ 85%]
test_trajectory_splitter.py ...............                                                                                                       [ 94%]
test_trajectory_stop_detector.py ..........                                                                                                       [100%]

File: movingpandas/tests/test_trajectory.py [hunk 2]
```diff
@@ -181,25 +181,30 @@ def test_add_direction_overwrite_raises_error(self):
...
     def test_add_speed_overwrite_raises_error(self):
         traj = make_traj([Node(0, 0), Node(6, 0, second=1)])
         traj.add_speed()
         with pytest.raises(RuntimeError):
             traj.add_speed()
 
+    def test_add_speed_with_name(self):
+        traj = make_traj([Node(0, 0), Node(6, 0, second=1)])
+        traj.add_speed(name="speed2")
+        assert "speed2" in traj.df.columns
```
Comment (anitagraser) [inline movingpandas/tests/test_trajectory.py:206]: We should also check that the values in a potentially already existing "speed" column are not changed and that the values in "speed2" are correct.

File: movingpandas/trajectory.py [hunk 3]
```diff
@@ -691,10 +693,13 @@ def add_speed(self, overwrite=False):
...
             Whether to overwrite existing speed values (default: False)
+        name : str
+            Name of the speed column (default: "speed")
         """
-        if SPEED_COL_NAME in self.df.columns and not overwrite:
-            raise RuntimeError('Trajectory already has speed values! Use overwrite=True to overwrite exiting values.')
-        self.df = self._get_df_with_speed()
+        self.speed_col_name = name
+        if self.speed_col_name in self.df.columns and not overwrite:
+            raise RuntimeError(f'Trajectory already has a column named {self.speed_col_name}! Use overwrite=True to overwrite exiting values.')
```
Comment (anitagraser) [inline movingpandas/trajectory.py:701]: We could point out to the user that, alternatively, they can specify a different column name.

File: movingpandas/trajectory.py [hunk 4]
```diff
@@ -23,7 +23,6 @@
                                         measure_distance_euclidean
 from .trajectory_plotter import _TrajectoryPlotter
 
-SPEED_COL_NAME = 'speed'
```
Comment (anitagraser) [inline movingpandas/trajectory.py:26]: Maybe better to keep this ...

File: movingpandas/trajectory.py [hunk 5]
```diff
@@ -247,7 +246,10 @@ def get_speed_column_name(self):
         -------
         string
         """
-        return SPEED_COL_NAME
+        if hasattr(self, "speed_col_name"):
+            return self.speed_col_name
+        else:
+            return "speed"
```
Comment (anitagraser) [inline movingpandas/trajectory.py:252]: ... and use it here

File: movingpandas/trajectory.py [hunk 6]
```diff
@@ -680,7 +682,7 @@ def add_distance(self, overwrite=False):
             raise RuntimeError('Trajectory already has distance values! Use overwrite=True to overwrite exiting values.')
         self.df = self._get_df_with_distance()
 
-    def add_speed(self, overwrite=False):
+    def add_speed(self, overwrite=False, name="speed"):
```
Comment (anitagraser) [inline movingpandas/trajectory.py:685]: ... and here

File: movingpandas/tests/test_trajectory.py [hunk 7]
```diff
@@ -7,7 +7,7 @@
 from shapely.geometry import Point, LineString
 from datetime import datetime, timedelta
 from fiona.crs import from_epsg
-from movingpandas.trajectory import Trajectory, DIRECTION_COL_NAME, SPEED_COL_NAME, DISTANCE_COL_NAME, MissingCRSWarning
```
Comment (anitagraser) [inline movingpandas/tests/test_trajectory.py:10]: Let's no hard-code the default. speed column name

**Review by anitagraser — APPROVED** [review]

**Pull request merged**